src/main/java/javaslang/collection/ScanImpl.java

```diff
@@ -0,0 +1,49 @@
+package javaslang.collection;
+
+import java.util.Objects;
+import java.util.function.BiFunction;
+
+class ScanImpl {
```

Comment:
The compiler flag `-Xlint:all -Werror` prevents compiling if auxiliary classes are accessed from outside of the class file.